sonic_sku_create/sonic_sku_create.py

```diff
@@ -0,0 +1,329 @@
+				return -1
+
+
+	def l2_mode(self,l2_sku) :
+		if not os.path.exists(DEFAULT_DEV_PATH+self.platform+"/"+l2_sku):
+			print ("Error - path:", l2_sku, " doesn't exist")
+		print( "Starting L2 mode configuration based on "+l2_sku+" SKU")	
+		cfg_db_str = "\n{\n\t\"DEVICE_METADATA\": {\n\t\t\"localhost\": {\n\t\t\t\"hostname\": \"sonic\"\n\t\t}\n\t}\n}\n"
+
+		subprocess.call("cat <<EOF | sudo config reload /dev/stdin -y" + cfg_db_str, shell=True)
+		subprocess.call("sudo sonic-cfggen -H --write-to-db ", shell=True)
+		subprocess.call("sonic-cfggen -t /usr/local/lib/python2.7/dist-packages/usr/share/sonic/templates/l2switch.j2 -p -k" + l2_sku + " | sudo config load /dev/stdin -y", shell=True ) 
+		subprocess.call("sudo config save -y ", shell=True)
+		subprocess.call("sudo systemctl restart swss ", shell=True)
+
+		print( "Finished L2 mode configuration based on "+l2_sku+" SKU")
+		
+def main():
+	parser = argparse.ArgumentParser(description='Create a new SKU',
+									version='1.0.0',
+									formatter_class=argparse.RawTextHelpFormatter)
+	parser.add_argument('-f', '--file', action='store', help='SKU definition file (default <cwd>/sku_def.xml)', default=None)
+	parser.add_argument('-b', '--base', action='store', help='SKU base definition  ', default=None)
+	parser.add_argument('-r', '--remove', action='store_true', help='Remove SKU folder')
+	parser.add_argument('-c', '--cmd', action='store', choices=['new_sku_only', 'l2_mode_only', 'new_sku_l2'], help='Choose action to preform (Generate a new SKU, Configure L2 mode, Both', default="new_sku_only")
+	parser.add_argument('-l2_sku_name', '--l2_sku_name', action='store', help='SKU name to be used in the L2 configuration mode', default=None)
+	parser.add_argument('-p', '--print', action='store_true', help='Print port_config.ini without creating a new SKU', default=False)
+	parser.add_argument('-vv', '--verbose', action='store_true', help='Verbose output', default=False)
+	args = parser.parse_args()
+	
+	l2_mode = False
+	sku_mode = False
+	sku_name = None
+	try:
+		sku = SkuCreate()
+		if args.cmd == "l2_mode_only":
+			l2_mode = True
+			sku_mode = False
+		elif args.cmd == "new_sku_l2":
+			l2_mode = True
+			sku_mode = True
+		else :
+			l2_mode = False
+			sku_mode = True		
+		
+		if args.file:
+			SKU_DEF_FILE = args.file
+		else:
+			SKU_DEF_FILE = os.getcwd() + '/sku_def.xml'
+		
+		sku.sku_def_parser(SKU_DEF_FILE,args.base)
+
+		if sku_mode :
+			if args.remove:
+				sku.remove_sku_dir()
+				return
+			sku.get_default_lanes()
+			sku.split_analyze()
+			sku.set_lanes()
+			if args.print:
+				sku.print_port_config()
+			else:
+				sku.create_sku_dir()
+				sku.create_port_config()
+				print ("Created a new sku ",sku.sku_name)
+			
+		if l2_mode : ##If No SKU name provided by -l2_sku_name then if -file exists, use the sku from the file otherwise use current SKU 
+			if args.l2_sku_name is None :
+				if args.file is None :
+					sku_name = sku.metadata['hwsku']
```

Comment:
>metadata [](start = 20, length = 8)

The script will be useful when we have an old image but we need a new SKU. Any old SKU in the image is not used if we generated a new SKU.

However, L2 mode in this script is not required.

Reply:
This is relevant when you have an old SKU and you want to configure it to L2 mode.
If the user will not indicate a specific SKU to configure as L2 mode nor a SKU def file will be provided, the script will use the current SKU to be used in the L2 mode.